### ISSUE TITLE
Fallback for failing custom action not working

### ISSUE THREAD

**Issue by HerrHorizontal** [HEPCompSim/DistCacheSim#30]
When running a simulation with an initial hit-rate at simulation start bigger than `0.0` and a sufficiently large average input-file size, e.g. 
```
sgbatch-sim --platform data/platform-files/sgbatch.xml --njobs 9 --output-file test.csv --hitrate 0.1 --insize 36000000000 --wrench-full-log
```
, the simulation fails. Following the logging of the job causing the program to fail
```
[2.00021333735493422878][sg01:action_execution_service_5015(10)] Got a [wrench::ActionExecutionServiceSubmitActionRequestMessage] message
[2.00021333735493422878][sg01:action_execution_service_5015(10)] Asked to run action streaming_job_6
[2.00021333735493422878][sg01:controller_action_executor_5093(32)] New Action Executor started to do action streaming_job_6
[2.00021333735493422878][sg01:controller_action_executor_5093(32)] Determining file sources for cache computation
[2.00034025906332679767][sg01:controller_action_executor_5093(32)] Evicting file infile_2_1 from storage service on host sg01
[2.00034047019734728678][sg01:controller_action_executor_5093(32)] Deleted file infile_2_1 at location simple_storage_0_5000:/
[2.00035539012534568926][sg01:controller_action_executor_5093(32)] Evicting file infile_4_2 from storage service on host sg01
[2.00035560125936617837][sg01:controller_action_executor_5093(32)] Deleted file infile_4_2 at location simple_storage_0_5000:/
[2.00037045716292816167][sg01:controller_action_executor_5093(32)] Evicting file infile_0_9 from storage service on host sg01
[2.00037066829694865078][sg01:action_execution_service_5015(10)] Got a [wrench::ActionExecutorDoneMessage] message
[2.00037066829694865078][sg01:action_execution_service_5015(10)] Sending action failure notification to 'mb5013'
[2.00037066829694865078][sg01:controller_action_executor_5093(32)] Action executor for action streaming_job_6 terminating!
[2.00037805798767021059][WMSHost:htcondor_central_manager_5033(13)] HTCondor Central Manager got a [wrench::ComputeServiceCompoundJobFailedMessage] message
[2.00037805798767021059][WMSHost:htcondor_central_manager_5033(13)] Uncaught exception std::runtime_error: Unexpected [wrench::ComputeServiceCompoundJobFailedMessage] message
```
it seems, that the issue is due to the fact, that the input-file can't be successfully cached and the action terminates. However, the failure message can't be successfully processed by the `HTCondorCentralManager`, so it never reaches the job manager and the action controller.

**Comment by HerrHorizontal** [comment]
@henricasanova do you think this might be an issue in WRENCH, or is it just a bad implementation in this simulator?

**Comment by henricasanova** [comment]
I'll look at this soon....

**Comment by henricasanova** [comment]
Ok. I did investigate, and yes! Somehow, in the WRENCH-2.0 rewrite we missed to add a 2-line simple "forward job failure notifications" code fragment to the HTCondorCentralManager.  So I just fixed that and it's now fixed in the WRENCH master branch. I just double-checked and your simulator now works. Thanks for finding that out. I'll add a test to the WRENCH unit tests as well, since obviously that case wasn't covered.
